src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1509,3 +1517,39 @@ def set_attention_slice(self, slice_size: Optional[int]):
 
         for module in modules:
             module.set_attention_slice(slice_size)
+
+    @classmethod
+    def _get_linked_pipelines(cls, task):
+        linked_classes_str = list(set([cls.__name__] + cls._linked_pipelines))
+        diffusers_library = importlib.import_module(__name__.split(".")[0])
+        linked_classes = [getattr(diffusers_library, c) for c in linked_classes_str if hasattr(diffusers_library, c)]
+        linked_classes = [c for c in linked_classes if c.task == task]
+        return linked_classes
+
+    @classmethod
+    def from_pipe(cls, pipeline):
+        pipeline_class = pipeline.__class__
+
+        if cls.__name__.startswith("Auto"):
+            pipeline_class_linked = pipeline_class._get_linked_pipelines(cls.task)
+            if len(pipeline_class_linked) == 0:
+                raise ValueError(
+                    f"can't find linked pipeline with task {cls.task} for pipeline class {pipeline_class}"
+                )
+            else:
+                pipeline_class = pipeline_class_linked[0]
+
+        model = pipeline_class(**pipeline.components)
+        return model
+
+
+class AutoPipelineForTextToImage(DiffusionPipeline):
```

Comment:
can we just have one `AutoDiffusionPipeline` and take a ` task` argument? 
@patrickvonplaten

e.g. 
```python
pipe_img2img = AutoDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", task="image-to-image")
pipe_txt2img =AutoDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", task="text-to-image")
pipe_inpaint = AutoDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5",task="inpaint")
```

Reply:
Hmm, I think it's nicer to have `AutoPipelineForText2Image` because otherwise `AutoDiffusionPipeline` is essentially the same as `DiffusionPipeline`